mobile/src/utils/utils.ts

```diff
@@ -1,10 +1,47 @@
 import { AppState } from '../reducers/app.reducer';
 import { Employee } from '../reducers/organization/employee.model';
 import { Optional } from 'types';
+import { DayModel, IntervalModel, ReadOnlyIntervalsModel } from '../reducers/calendar/calendar.model';
+import { CalendarEvent } from '../reducers/calendar/calendar-event.model';
 
 //----------------------------------------------------------------------------
 export function getEmployee(state: AppState): Optional<Employee> {
     return (state.organization && state.userInfo && state.userInfo.employeeId) ?
         state.organization.employees.employeesById.get(state.userInfo.employeeId) :
         undefined;
 }
+
+//----------------------------------------------------------------------------
+function getVacationIntervals(intervals: IntervalModel[], exclude: CalendarEvent[] = []): IntervalModel[] {
+    return intervals
+        .filter(interval => interval.calendarEvent.isVacation)
+        .filter(interval => !exclude.find(event => interval.calendarEvent.calendarEventId === event.calendarEventId));
+}
+
+//----------------------------------------------------------------------------
+export function isIntersectingAnotherVacation(startDay: DayModel | undefined,
+                                              endDay: DayModel | undefined,
+                                              intervals: ReadOnlyIntervalsModel,
+                                              exclude: CalendarEvent[] = []): boolean {
+    if (!startDay) {
+        return false;
+    }
+
+    if (!endDay) {
+        const intervalsForStartDay = intervals.get(startDay.date);
```

Comment:
As I understood, `intervals.get(date)` returns all intervals that starts from specified date. Am I right?

Reply:
Huh, the same question I asked. In fact, intervals are organized so that all days containing in the interval are keys for that interval in the ‘intervals’ structure. So, the code is good.

Reply:
So, if there is an event from 01-01-2018 till 5-01-2018, dictionary will contain entries:
```ts
{
    '01-01-2018': [ event ],
    '02-01-2018': [ event ],
    '03-01-2018': [ event ],
    '04-01-2018': [ event ],
    '05-01-2018': [ event ]
}
```
If it is so, it will work.

Reply:
Exactly as you described.